### PR TITLE
Fix capture of `GlobalPhase` and `Identity` when they act on no wires

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

When writing tests for [(#837)](https://github.com/PennyLaneAI/catalyst/pull/837), I found we weren't capturing `Identity` and `GlobalPhase` when they act on zero wires correctly.

**Description of the Change:**

Fix the capture of `GlobalPhase` and `Identity` when they act on no wires.

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-70350]

### CODE REVIEW

File: doc/releases/changelog-dev.md [hunk 1]
```diff
@@ -236,6 +236,8 @@
 
 <h3>Bug fixes 🐛</h3>
 
+* `qml.GlobalPhase` and `qml.I` can now be captured when acting on no wires.
```
Comment (albi3ro) [inline doc/releases/changelog-dev.md:239]:
```suggestion
* `qml.GlobalPhase` and `qml.I` can now be captured when acting on no wires.
  [(#6060)](https://github.com/PennyLaneAI/pennylane/pull/6060)
```

**Review by dime10 — APPROVED** [review]
Thanks for the fix 👍 

(Related issue regarding the documentation for GlobalPhase: https://github.com/PennyLaneAI/pennylane/issues/5820)

File: pennylane/capture/primitives.py [hunk 2]
```diff
@@ -190,10 +190,11 @@ def _(*args, **kwargs):
             return type.__call__(operator_type, *args, **kwargs)
         n_wires = kwargs.pop("n_wires")
 
+        split = None if n_wires == 0 else -n_wires
         # need to convert array values into integers
         # for plxpr, all wires must be integers
-        wires = tuple(int(w) for w in args[-n_wires:])
-        args = args[:-n_wires]
+        wires = tuple(int(w) for w in args[split:])
+        args = args[:split]
```
Comment (dwierichs) [inline pennylane/capture/primitives.py:197]: Wow, I did not know about `None` indexing into `tuple` / `list` :sweat_smile: 
Could also do `split = len(args) - n_wires`, but no need to change!

File: tests/capture/test_operators.py [hunk 3]
```diff
@@ -193,27 +193,64 @@ def test_parametrized_op():
...
+        assert len(eqn.invars) == 3  # The rotation parameter and the two wires
+        assert jaxpr.jaxpr.invars == eqn.invars
 
-    assert len(q) == 1
-    qml.assert_equal(q.queue[0], qml.PauliRot(2.5, "XY", (3, 4)))
+        with qml.queuing.AnnotatedQueue() as q:
+            jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 2.5, 3, 4)
+
+        assert len(q) == 1
+        qml.assert_equal(q.queue[0], qml.PauliRot(2.5, "XY", (3, 4)))
```
Comment (dwierichs) [inline tests/capture/test_operators.py:218]: Got to love git diffing

**Review by dwierichs — APPROVED** [review]
Nice catch, thanks @albi3ro

File: tests/capture/test_operators.py (hunk 3 above)
Comment (albi3ro) [reply tests/capture/test_operators.py:218]: 😢

**Pull request merged**